lib/AST/AlignTokens.cpp

```diff
@@ -563,42 +480,48 @@ static bool MergeToken(TokenImpl *parsed, PrintedTokenImpl *printed,
 class Matcher {
  private:
   ASTImpl &ast;
-  PrintedTokenRangeImpl &range;
-  TokenImpl * const first_parsed;
-
-  // Exclusive upper bound.
-  TokenImpl * const last_parsed;
+  PrintedTokenRangeImpl &parsed_range;
+  PrintedTokenRangeImpl &printed_range;
+  PrintedTokenImpl * const first_parsed;
+  PrintedTokenImpl * const last_parsed;  // Exclusive, but can be read.
+  PrintedTokenImpl * const first_printed;
+  PrintedTokenImpl * const last_printed;  // Exclusive, but can be read.
 
   std::vector<std::unique_ptr<Region>> parsed_regions;
   std::vector<std::unique_ptr<Region>> printed_regions;
```

Comment:
Wrap in on structure and then have
```
X parsed;
X printed;
```

Reply:
This is a good suggestion; it used to be that the types were different (`TokenImpl` and `PrintedTokenImpl`).

Reply:
I've worked a bit of this in in places, but there'll be room for improvement in the future too.